addons/web_dashboard/models/res_users.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+
+from datetime import datetime, timedelta
+
+from openerp import models, api, _
+from openerp.exceptions import RedirectWarning
+from openerp.tools.misc import DEFAULT_SERVER_DATETIME_FORMAT
+
+
+class Users(models.Model):
+    _inherit = 'res.users'
+
+    @api.model
+    def web_dashboard_create_users(self, emails, optional_message=""):
+        existing_users = self.search(['&', ('login', 'in', emails), '|', ('active', '=', True), ('active', '=', False)])
+
+        # Processing existing users
+        for user in existing_users:
+            if not user.login_date and not user.signup_valid:
+                user.action_reset_password()
+            else:
+                # Already a user, reactivated in case it is deactivated
+                user.active = True
+        new_emails = set(emails) - set(existing_users.mapped('login'))
+        template_user_id = self.env['ir.config_parameter'].get_param('auth_signup.template_user_id', 'False')
+
+        # Processing new email addresses : create new users (Used template user)
+        for email in new_emails:
+            default_values = {'login': email, 'name': email, 'email': email, 'active': True}
```

Comment:
You can take look at this, because of user have email id in name i provided a way at sign up, so user can change his/her name at sign up(just one time)
https://github.com/odoo/odoo/pull/7288/files#diff-fe544dce57f31c7962edb0ce697b5431R198